mysql/ci/mysql.rake

```diff
@@ -14,31 +14,50 @@ container_port = 13_306
 namespace :ci do
   namespace :mysql do |flavor|
     task before_install: ['ci:common:before_install'] do
-      `docker kill $(docker ps -q --filter name=dd-test-mysql) || true`
-      `docker rm $(docker ps -aq --filter name=dd-test-mysql) || true`
+      `docker kill $(docker ps -q --filter name=#{container_name}_slave) || true`
+      `docker rm $(docker ps -aq --filter name=#{container_name}_slave) || true`
+      `docker kill $(docker ps -q --filter name=#{container_name}_master) || true`
+      `docker rm $(docker ps -aq --filter name=#{container_name}_master) || true`
     end
 
     task :install do
       Rake::Task['ci:common:install'].invoke('mysql')
-      sh %(docker run -p #{container_port}:3306 --name #{container_name} -e MYSQL_ROOT_PASSWORD=datadog -d mysql:5.7)
+      sh %(docker run -p #{container_port}:3306 --name #{container_name}_master \
+           -e MYSQL_ALLOW_EMPTY_PASSWORD=1 -d bergerx/mysql-replication:#{mysql_version})
     end
 
     task before_script: ['ci:common:before_script'] do
       Wait.for 13_306
       count = 0
-      logs = `docker logs dd-test-mysql 2>&1`
+      logs = `docker logs #{container_name}_master 2>&1`
       puts 'Waiting for MySQL to come up'
       until count == 20 || logs.include?('MySQL init process done. Ready for start up')
         sleep_for 2
-        logs = `docker logs dd-test-mysql 2>&1`
+        logs = `docker logs #{container_name}_master 2>&1`
         count += 1
       end
       if logs.include? 'MySQL init process done. Ready for start up'
         puts 'MySQL is up!'
       end
 
-      sh %(docker run -it --link dd-test-mysql:mysql --rm mysql:5.7 \
-           sh -c 'exec mysql -h"$MYSQL_PORT_3306_TCP_ADDR" -P"MYSQL_PORT_3306_TCP_PORT" -uroot -p"$MYSQL_ENV_MYSQL_ROOT_PASSWORD" \
+      sh %(docker run -p #{container_port + 1}:3306 --name #{container_name}_slave \
+           -e MYSQL_ALLOW_EMPTY_PASSWORD=1 --link #{container_name}_master:master \
+           -d bergerx/mysql-replication:#{mysql_version})
+      Wait.for 13_307
+      count = 0
+      logs = `docker logs #{container_name}_slave 2>&1`
+      puts 'Waiting for MySQL to come up'
+      until count == 20 || logs.include?('MySQL init process done. Ready for start up')
+        sleep_for 2
+        logs = `docker logs #{container_name}_slave 2>&1`
+        count += 1
+      end
+      if logs.include? 'MySQL init process done. Ready for start up'
+        puts 'MySQL is up!'
```

Comment:
could we throw an explicit error if the mysql init process is still not ready at this point?